[AI] Display user-friendly errors

millerf opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes https://github.com/Automattic/jetpack/issues/28571 and https://github.com/Automattic/jetpack/issues/28570

## Proposed changes:
* Displays the error sent by the back-end if needed

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Patch this to your sandbox
* Patch D99484-code
* switch production `bin/jetpack-downloader switch jetpack production`
* Use some personality names for image generation or modify OPENAI_TOKEN 
* Generate soem AI content
* You should see some user-friendly errors

millerf committed:
Display user_message if available

millerf committed:
Changelog

artpi commented on projects/plugins/jetpack/extensions/blocks/ai-image/edit.js:
```diff
@@ -54,13 +54,17 @@ function getImagesFromOpenAI(
-				)
-			);
+		.catch( e => {
+			if ( e.data?.user_message ) {
```
Frankly I think we should just use the regular `message` payload instead of introducing a new field. We can keep the secret stuff in logstash

artpi replied:
Taht would be more consistent with the WP_Error format instead of introducing custom fields

millerf replied:
I feel you, but the idea was to prevent [those errors](fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Sbcranv%2Sbcranv.cuc%23104-og) to be thrown into the user's face and [get the default error instead](https://github.com/Automattic/jetpack/blob/1cf7d4b0c478ed1287599e1e05390b56af6a65c8/projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js#L223)...

millerf replied:
But I can take it the other way around as well and fit the WP_Error format. I'll work on that quickly.

artpi replied:
So on backend we could just catch the openai errors and throw a different one with the user-facing error message

millerf replied:
YEs yes.

artpi reviewed (dismissed):
I still prefer more passing the message in the regular error message field instead of user_message, but we can discuss this later.
This is great for now

millerf committed:
Display the e.message instead of custom e.data.user_message

millerf committed:
Remove unused code

millerf commented:
@artpi I reworked the back and the front to throw the error using the WP_error `message` attribute. I also added the original error in the `data` field when needed, for debugging ease.

<img width="200" alt="Screenshot 2023-01-27 at 16 51 53" src="https://user-images.githubusercontent.com/790558/215129896-3de22059-9e10-428d-8617-2ffff1511de0.png">
<img width="200" alt="Screenshot 2023-01-27 at 16 52 18" src="https://user-images.githubusercontent.com/790558/215129904-cf43ce7b-184d-411c-94dd-c0ae48ca54bf.png">

millerf committed:
Merge branch 'trunk' into add/open-ai-display-user-error

artpi reviewed (approved):
Works!
